code/src/java/pcgen/core/Description.java

```diff
@@ -359,11 +348,7 @@ else if (str.equals("%"))
 		}
 		if ( theVariables != null )
 		{
-			for ( final String var : theVariables )
-			{
-				buf.append(Constants.PIPE);
-				buf.append(var);
-			}
+			theVariables.stream().map(var -> Constants.PIPE + var).forEach(buf::append);
```

Comment:
This does 2 StringBuffers because you have a PIPE+var.  So this will be a huge performance hit that I don't think is valuable just to do .stream()